### ISSUE TITLE
Paragraphs in longtable don't work in Latex output

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1580]
There is another problem when you have multiple paragraphs in one table cell, for example like this:

```
+----+----+
| 00 | X  |
+----+----+
| 01 | X  |
+----+----+
| 02 | P1 |
|    |    |
|    | P2 |
+----+----+
| 03 | X  |
+----+----+
| 04 | X  |
...
| 30 | X  |
+----+----+
```

This works in HTML, and in Latex as long as there are not over 30 rows. P1 and P2 are separate paragraphs on separate lines.

But above 30 rows, the latex writer switches from `tabulary` to `longtable`. For some reason that causes the paragraphs P1 and P2 to get merged onto the same line.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1580
- Originally reported by: Peter Suter
- Originally created at: 2014-09-27T15:05:12.514

**Comment by shimizukawa** [comment]
_From Peter Suter on 2014-09-29 08:48:21+00:00_

The `self.table.has_problematic` mode of `sphinx/writers/latex.py` (i.e. changing `colspec` from `l` to `p`) seems to help. I'm not sure what the downsides are though.

Should `if self.table: self.table.has_problematic = True` be added to `LaTeXTranslator.visit_paragraph`?

Or should `LaTeXTranslator.depart_table` just be changed to always use `p` for `self.table.longtable`?

Or maybe there is a better solution?

[tex.stackexchange](http://tex.stackexchange.com/questions/2441/how-to-add-a-forced-line-break-inside-a-table-cell) has many more or less complicated suggestions for how to force line breaks in table cells...

**Comment by shimizukawa** [comment]
_From Peter Suter on 2014-10-08 09:18:09+00:00_

(As a workaround the [.. tabularcolumns::](http://sphinx-doc.org/markup/misc.html?highlight=tabularcolumns#directive-tabularcolumns) directive can be used to override the broken default colspec.)

**Comment by tk0miya** [comment]
Fixed at 2be21e3.
Thank you for reporting
